core/platform.js

```diff
@@ -191,24 +188,18 @@ Platform.prototype.shutdown = function(flag) {
     }
 
     // Unload user modules
-    for (var i = 0; i < this.loadedModules.length; i++) {
-        if (this.loadedModules[i].unload) {
-            this.loadedModules[i].unload();
-        }
-        this.loadedModules[i] = null;
+    while (this.loadedModules.length > 0) {
+        this.modulesLoader.unloadModule(this.loadedModules[0]);
+        this.loadedModules.splice(0, 1);
     }
-    this.loadedModules = [];
-
+    
     // Unload core modules
-    for (var i = 0; i < this.coreModules.length; i++) {
-        if (this.coreModules[i].unload) {
-            this.coreModules[i].unload();
-        }
-        this.coreModules[i] = null;
+    while (this.coreModules.length > 0) {
+        this.coreLoader.unloadCoreModule(this.coreModules[0]);
+        this.coreModules.splice(0, 1);
     }
-    this.coreModules = [];
 
-    this.config.saveConfig();
+    this.config.saveSystemConfig();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["StatusFlag" is not defined.](https://www.codacy.com/app/theprojectshop/Kassy/file/2530104146/issues/source?bid=3282987&fileBranchId=3282997#l203)